src/Models/Behaviors/HasRevisions.php

```diff
@@ -36,14 +36,22 @@ public function scopeMine($query)
      */
     public function revisionsArray()
     {
-        return $this->revisions->map(function ($revision, $index) {
-            return [
-                'id' => $revision->id,
-                'author' => $revision->user->name ?? 'Unknown',
-                'datetime' => $revision->created_at->toIso8601String(),
-                'label' => $index === 0 ? twillTrans('twill::lang.publisher.current') : '',
-            ];
-        })->toArray();
+        $currentRevision = null;
+
+        return $this->revisions
+            ->map(function ($revision, $index) use (&$currentRevision) {
+                if (!$currentRevision && !$revision->isDraft()) {
+                    $currentRevision = $revision;
+                }
+
+                return [
+                    'id' => $revision->id,
+                    'author' => $revision->user->name ?? 'Unknown',
+                    'datetime' => $revision->created_at->toIso8601String(),
+                    'label' => $currentRevision === $revision ? twillTrans('twill::lang.publisher.current') : '',
```

Comment:
I wonder how "safe" this comparison is: `$currentRevision === $revision`

Perhaps we can compare the id's?

Reply:
I'm not sure I follow. Can you expand on this a bit more?

Here, we start from a fixed list of object references ($this->revisions), and try to identify which one should have the `Current` label. In my view, comparing object references is not a problem.

FWIW, there a 2 tests covering this method: [(1)](https://github.com/area17/twill/pull/1725/files/d8b4f559c5ef20019b17fdf4f2dbf2c1a85d0c54#diff-695f6fd51bb71c5db4574813ec07caefc86109a094bcd60b28b482c17776c81dR119) [(2)](https://github.com/area17/twill/pull/1725/files/d8b4f559c5ef20019b17fdf4f2dbf2c1a85d0c54#diff-695f6fd51bb71c5db4574813ec07caefc86109a094bcd60b28b482c17776c81dR189)